test/Microsoft.AspNetCore.Mvc.Abstractions.Test/ModelBinding/ModelStateDictionaryTest.cs

```diff
@@ -684,6 +703,31 @@ public void GetFieldValidity_ReturnsValid_IfAllKeysAreValid()
             Assert.Equal(ModelValidationState.Valid, validationState);
         }
 
+        [Fact]
+        public void AddModelError_WithException_AddsTooManyModelErrorsException_WhenMaxErrorsIsReached()
+        {
+            // Arrange
+            var expected = "The maximum number of allowed model errors has been reached.";
+            var dictionary = new ModelStateDictionary
+            {
+                MaxAllowedErrors = 2
+            };
+
+            dictionary.AddModelError("key1", new Exception());
+            dictionary.AddModelError("key2", "error4");
+            dictionary.AddModelError("key3", new Exception());
+
+            // Act and Assert
+            Assert.True(dictionary.HasReachedMaxErrors);
+            Assert.Equal(2, dictionary.ErrorCount);
+            var error = Assert.Single(dictionary[string.Empty].Errors);
+            Assert.IsType<TooManyModelErrorsException>(error.Exception);
+            Assert.Equal(expected, error.Exception.Message);
+
+            // TooManyModelErrorsException added instead of key3 exception.
```

Comment:
Once `key3` is attempted, the dictionary is already full of errors. Add a check that `key2` doesn't exist in the dictionary either.

Reply:
Why wouldn't `key2` exist in the dictionary? 2 errors are allowed. Are all errors dropped if max is reached?

Reply:
The limit includes the `TooManyModelErrorsException` entry. In this test, that means one error for `"key1"` and one (containing the `TooManyModelErrorsException`) for `string.Empty`.